cloud/services/subnets/subnets.go

```diff
@@ -66,16 +73,25 @@ func (s *Service) Reconcile(ctx context.Context) error {
 
 			subnet.Role = subnetSpec.Role
 			subnet.Name = existingSubnet.Name
-			subnet.CidrBlock = existingSubnet.CidrBlock
+			subnet.CIDRBlocks = existingSubnet.CIDRBlocks
 			subnet.ID = existingSubnet.ID
 
 		case !s.Scope.IsVnetManaged():
 			return fmt.Errorf("vnet was provided but subnet %s is missing", subnetSpec.Name)
 
 		default:
+
 			subnetProperties := network.SubnetPropertiesFormat{
-				AddressPrefix: to.StringPtr(subnetSpec.CIDR),
+				AddressPrefixes: &subnetSpec.CIDRs,
 			}
+
+			// workaround needed to avoid SubscriptionNotRegisteredForFeature for feature Microsoft.Network/AllowMultipleAddressPrefixesOnSubnet.
```

Comment:
Any idea on when this will no longer be feature flagged?

Reply:
No idea, I couldn't find any documentation on it unfortunately